purchase_product_weight/tests/test_purchase_weight.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+
+from odoo.tests.common import TransactionCase
+import time
+
+
+class PurchaseOrder(TransactionCase):
+
+    def setUp(self):
+        super(PurchaseOrder, self).setUp()
+        self.product_model = self.env['product.template']
+        self.product_template_on_weight = self.product_model.create({
+            'name': 'Product on weight',
+            'type': 'product',
+            'uom_id': self.env.ref('product.product_uom_meter').id,
+            'uom_po_id': self.env.ref('product.product_uom_meter').id,
+            'list_price': 50.0,
+            'taxes_id': [(6, 0, self.env['account.tax'].search(
+                [('type_tax_use', '=', 'sale')], limit=1).ids)],
+            'supplier_taxes_id': [(6, 0, self.env['account.tax'].search(
+                [('type_tax_use', '=', 'purchase')], limit=1).ids)],
+            'compute_price_on_weight': True,
+            'weight': 0.15,
+            'seller_ids': [
+                (0, 0, {
+                    'name': self.env.ref('base.res_partner_3').id,
+                    'price': 5.0,
+                    'min_qty': 0.0,
+                })
+            ]
+        })
+        self.product_on_weight = self.env['product.product'].search([
+            ('product_tmpl_id', '=', self.product_template_on_weight.id)
+        ], limit=1)
+
+    def test_purchase_order_on_weight(self):
+        purchase_order = self.env['purchase.order'].create({
+            'partner_id': self.env.ref('base.res_partner_3').id,
+            'date_order': time.strftime('%Y-%m-%d'),
```

Comment:
@sergiocorato IMO Here Use Server Datetime

Reply:
@Nikul-Chaudhary you mean, `fields.Date.context_today(self)`?

Reply:
:+1: